datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaTopicPartitionTracker.java

```diff
@@ -32,8 +33,8 @@
   private final String _consumerGroupId;
 
   private final Map<String, Set<Integer>> _topicPartitions = new ConcurrentHashMap<>();
-
-  private final Map<String, Map<Integer, Long>> _consumerOffsets = new ConcurrentHashMap<>();
+  private final Map<String, Map<Integer, Long>> _consumedOffsets = new ConcurrentHashMap<>();
```

Comment:
I was considering adding a separate `ConsumerOffsets` class to hold both consumed offset and committed offset and use that in the nested dictionary. But synchronizing the writes to the nested dictionaries was a not easy. `onPartitionsPolled` and `onOffsetsCommitted` calls would hold locks on the dictionaries and it would make this code bug prone and non-efficient. So I thought it would be better for the client-side to merge these dictionaries (if necessary for presentation purposes).